Fix rotated secrets JSON parsing when parse-json-secrets is enabled

deanshak opened the pull request:
- Handle rotatedSecret.value when it's already an object by stringifying it before parsing
- Update parseJson function to handle both objects and strings
- Improve error messages to include secret name for better debugging
- Fixes issue where rotated secrets with parse-json-secrets=true would fail to export individual fields

deanshak committed:
Fix rotated secrets JSON parsing when parse-json-secrets is enabled

- Handle rotatedSecret.value when it's already an object by stringifying it before parsing
- Update parseJson function to handle both objects and strings
- Improve error messages to include secret name for better debugging
- Fixes issue where rotated secrets with parse-json-secrets=true would fail to export individual fields

deanshak committed:
Fix rotated secrets JSON parsing when parse-json-secrets is enabled

- Handle rotatedSecret.value when it's already an object by stringifying it before parsing (only when parseJsonSecrets=true)
- Update parseJson function to handle both objects and strings
- Improve error messages to include secret name for better debugging
- Preserve backward compatibility: old behavior maintained when parseJsonSecrets=false
- Bump version to v1.1.4

Fixes issue where rotated secrets with parse-json-secrets=true would fail to export individual fields

tuvia-akeyless reviewed (approved):
LGTM

amirmao commented:
> * Handle rotatedSecret.value when it's already an object by stringifying it before parsing
> * Update parseJson function to handle both objects and strings
> * Improve error messages to include secret name for better debugging
> * Fixes issue where rotated secrets with parse-json-secrets=true would fail to export individual fields

nice work :) 
Can you please add tests in `tests/secrets.test.js` for the use cases?

deanshak committed:
Add tests for rotated secrets with parse-json-secrets

- Add test for parse-json-secrets=true with prefix (verifies the fix)
- Add test for parse-json-secrets=true without prefix (default prefix)
- Add test for parseJsonSecrets=false (backward compatibility verification)
- All 25 tests pass, covering the fix and backward compatibility

deanshak commented:
Thanks @amirmao & @tuvia-akeyless 
I’ve added tests for the use cases.

amirmao reviewed (approved):
LGTM

deanshak commented:
@amirmao It looks like it still requires an additional reviewer and someone with permissions to merge it.

amirmao commented:
> @amirmao It looks like it still requires an additional reviewer and someone with permissions to merge it.

for some reason @tuvia-akeyless user is a member of akeyless community?  @omriezra please approve the PR
